src/UserManager.ts

```diff
@@ -263,15 +263,42 @@ export class UserManager {
     }
 
     protected async _useRefreshToken(state: RefreshState): Promise<User> {
-        const response = await this._client.useRefreshToken({
-            state,
-            timeoutInSeconds: this.settings.silentRequestTimeoutInSeconds,
+        const refreshUser = async (): Promise<User> => {
+            const response = await this._client.useRefreshToken({
+                state,
+                timeoutInSeconds: this.settings.silentRequestTimeoutInSeconds,
+            });
+            return new User({ ...state, ...response });
+        };
+
+        if (!navigator.locks) {
+            // Legacy option for older browser which don't support `navigator.locks`.
+            const user = await refreshUser();
+            await this.storeUser(user);
+            this._events.load(user);
+            return user;
+        }
+
+        const broadcastChannel = new BroadcastChannel(`refresh_token_${state.refresh_token}`);
+        let user : User | null = null;
+
+        broadcastChannel.addEventListener("message", (event : MessageEvent<User>) => {
+            user = event.data;
         });
-        const user = new User({ ...state, ...response });
 
-        await this.storeUser(user);
-        this._events.load(user);
-        return user;
+        return await navigator.locks.request(
+            `refresh_token_${state.refresh_token}`,
+            async () => {
+                if (!user) {
+                    user = await refreshUser();
+                }
+
+                broadcastChannel.postMessage(user);
```

Comment:
The last two steps are unnecessary:
- A creates a broadcast channel and listen to it
- A request a lock and do the job
- B creates a broadcast channel and listen to it
- B request a lock and waits
- A finish the job and broadcast the user
- B receives the broadcast and keeps the user
- A releases the lock
- B proceed with the job but already has the user (refreshUser() is skipped)
- **B broadcast the user**
- **A receives the broadcast**

The broadcast should only occur if a refresh was made.
So it should be included in the previous `if`.